packages/browser-agent-core/src/features/page-view-timing/aggregate/index.js

```diff
@@ -122,28 +123,23 @@ export class Aggregate extends FeatureBase {
   }
 
   /**
-   * Add the time of _document visibilitychange to hidden_ to the next PVT harvest.
+   * Add the time of _document visibilitychange to hidden_ to the next PVT harvest == NRDB pageHide attr.
    * @param {number} timestamp
    */
-  updateSessionEnd(timestamp) {
-    if (!this.userSessionEnded) { // TO DO: stage 2 - we don't want to capture this timing twice on page navigating away, but it should run again if we return to page and away *again*
-      this.addTiming('pageHide', timestamp, null, true)
-      this.userSessionEnded = true
+  endCurrentSession(timestamp) {
+    if (!this.curSessEndRecorded) { // TO DO: stage 2 - we don't want to capture this timing twice on page navigating away, but it should run again if we return to page and away *again*
```

Comment:
For Phase 2... Does this become something like... page is restored from cache or pageShow event, set curSessEndRecorded = false again?

Reply:
yeah, along with whatever else. CWV reset their internal metrics on pageshow.persisted